In [50]:
import numpy as np
import pandas as pd

import scipy.stats as stats

In [4]:
pesos = [100, 150, 150, 200, 250, 300, 325, 400,415, 500, 600, 1000]

**Media**

In [48]:
pesos_media = np.mean(pesos)
pesos_media

365.83333333333331

**mediana**

In [7]:
np.median(pesos)

312.5

**Moda**

In [51]:
stats.mode(pesos)

ModeResult(mode=array([150]), count=array([2]))

**Rango**

In [9]:
np.max(pesos) - np.min(pesos)

900

**Cuartiles y IQR**

In [54]:
stats.mstats.mquantiles(pesos)

array([ 172.5 ,  312.5 ,  461.75])

La funcion `pandas.qcut` convierte una serie en una lista de N cuantiles. Si le pasamos 4, tenemos los cuartiles. `qcut` devuelve una serie categórica nueva con los cuantiles como categorias

In [15]:
pesos_cuartiles = pd.qcut(pesos, 4)

pesos_cuartiles.categories

IntervalIndex([(99.999, 187.5], (187.5, 312.5], (312.5, 436.25], (436.25, 1000.0]]
              closed='right',
              dtype='interval[float64]')

Aqui se ve que los intervalos que separan los pesos en 4 bloques son:

`Categories (4, interval[float64]): [(99.999, 187.5] < (187.5, 312.5] < (312.5, 436.25] < (436.25, 1000.0]]`

con 187.5 siendo Q1 y 436.25 siendo Q3

Es importante notar que hay diversas formas de obtener los cuartiles, de aqui la diferencia con el método calculado a mano

In [22]:
IQR = 436.25 - 187.5
IQR

248.75

** Desviacion standard**

In [49]:
pesos_std = np.std(pesos)
pesos_std

239.67715554238561

** Coeficiente de Simetría**

El coeficiente de simetria (skewness) se define como el tercer momento dividido por la desviacion standard al cubo, tiene la formula siguiente: 

$$\frac{1}{N} * \frac{\sum_{n=1}^{n} (Xi-\bar{X})^{3}}{\sigma^3}$$

In [31]:
pow3 = lambda x: x*x*x

In [39]:
tercer_momento = lambda x: pow3(x - pesos_media)

In [45]:
simetria_pesos = sum(map(tercer_momento, pesos)) / (12*pow3(pesos_std))
simetria_pesos

1.3623858394083475

Una mejor forma es usar la funcion `scipy.stats.skew`

In [41]:
stats.skew(pesos)

1.3623858394083481

**Coeficiente de curtosis**

El coeficiente de oblicuidad o curtosis se define como el cuarto momento dividido por la desviacion estandard a la cuarta, tiene la formula siguiente: 

$$\frac{1}{N} * \frac{\sum_{n=1}^{n} (Xi-\bar{X})^{4}}{\sigma^4} - 3$$

Generalmente se le resta 3 al cuarto momento, ya que una distribución normal perfecta tiene un coeficiente de curtosis de 3

In [44]:
pow4 = lambda x: x*x*x*x
cuarto_momento = lambda x: pow4(x - pesos_media)

In [47]:
curtosis_pesos = sum(map(cuarto_momento, pesos)) / (12*pow4(pesos_std))  - 3
curtosis_pesos

1.4285722765161841

In [55]:
stats.kurtosis(pesos)

1.4285722765161841